### PR TITLE
Add --stop-on-failure option

### PR DESCRIPTION
**Pull request by Chemaclass**
## 📚 Description

Issue: https://github.com/TypedDevs/bashunit/issues/144 

## 🔖 Changes

Add a new option `--stop-on-failure` that will force to stop the runner right after encountering one failing test.

> Inspired by PHPUnit: https://docs.phpunit.de/en/10.3/configuration.html#the-stoponfailure-attribute

### CODE REVIEW

**Review by Tito-Kati — CHANGES_REQUESTED** [review]
It needs to be documented in the --help

File: bashunit [hunk 1]
```diff
@@ -40,6 +40,10 @@ while [[ $# -gt 0 ]]; do
       SIMPLE_OUTPUT=false
       shift
       ;;
+    --stop-on-failure)
```
Comment (Tito-Kati) [inline bashunit:43]: -S ?

File: src/env_configuration.sh [hunk 2]
```diff
@@ -20,3 +20,6 @@ fi
 if [[ -z "$SIMPLE_OUTPUT" ]]; then
   SIMPLE_OUTPUT=$_DEFAULT_SIMPLE_OUTPUT
 fi
+if [[ -z "$STOP_ON_FAILURE" ]]; then
+  STOP_ON_FAILURE=$_DEFAULT_STOP_ON_FAILURE
```
Comment (khru) [inline src/env_configuration.sh:24]: @Chemaclass when you merge this PR add this variable to the `.env.testing`

**Review by khru — APPROVED** [review]

**Review by fabriziofs — APPROVED** [review]

File: src/env_configuration.sh (hunk 2 above)
Comment (Chemaclass) [reply src/env_configuration.sh:25]: I added it to `.env.example`

**Pull request merged**